examples/self_encryption/main.rs

```diff
@@ -0,0 +1,138 @@
+// licence you accepted on initial access to the Software (the "Licences").
+//
+// By contributing code to the MaidSafe Software, or to this project generally, you agree to be
+// bound by the terms of the MaidSafe Contributor Agreement, version 1.0, found in the root
+// directory of this project at LICENSE, COPYING and CONTRIBUTOR respectively and also
+// available at: http://www.maidsafe.net/licenses
+//
+// Unless required by applicable law or agreed to in writing, the MaidSafe Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, either express or implied.
+//
+// See the Licences for the specific language governing permissions and limitations relating to
+// use of the MaidSafe
+// Software.
+
+//! Implementation of a simple commandline for Self-Encryption
+
+extern crate self_encryption;
+
+use std::path;
+use std::fs;
+use std::io::*;
+// use tempdir::TempDir as TempDir;
+
+macro_rules! iotry {
+    ($e:expr) => (match $e { Ok(v) => v, Err(e) => panic!("{}", e), })
+}
+
+fn usage() {
+  println!("Usage: self-encrypt or decrypt a local file");
+  println!("       se -e <source_file>");
+  println!("       se -d <source_file>");
+  println!("       note: use absolute paths");
+}
+
+pub struct MyStorage<'b> {
+  source_path : &'b path::Path,
+  name : &'b str;
+  dir : &'b path::Path,
+  destination_path : &'b path::Path
+}
+
+
+impl MyStorage {
+  pub fn new(filepath_ : &str) -> MyStorage {
+    MyStorage { source_path : match path::Path::new(filepath_) {
+    			  Ok(x) => x,
+    			  Err(e) => panic!("Could not parse source file. {}", e)
+                },
+    			name_ : match source_path.file_name().unwrap_or("default"){
+    			  Ok(x) => x,
+    			  Err(e) => panic!("Source file is not a file. {}", e)
+    			},
+    			destination : match source_path.parent()
+    								           .unwrap()
+    									       .join(path::Path::new(&name_))
+    									       .join("datamap"),
+     			dir : match path().join(path::Path::new(&file_name));{
+        		  Ok(dir) => dir,
+                  Err(why) => 
+                    panic!("couldn't create directory: {}", why)
+                }
+              }
+  }
+}
+
+impl self_encryption::Storage for MyStorage {
+  fn get(&self, name: Vec<u8>) -> Vec<u8> {
+    let file_name = String::from_utf8(name).unwrap();
+    let file_path = self.temp_dir.path().join(path::Path::new(&file_name)); 
+    let mut f = match fs::File::open(&file_path) {
+        // The `desc` field of `IoError` is a string that describes the error
+        Err(why) => panic!("on get couldn't open: {}", why),
+        Ok(file) => file,
+    };
+    let mut s = String::new();
+    //f.read_to_string(&mut s); put f into a string
+    match f.read_to_string(&mut s){
+        Err(why) => panic!("on get couldn't read: {}", why),
+        Ok(_) => print!("contains:\n{}", s),
+    }
+    s.into_bytes()
+  }
+
+  #[allow(unused_must_use)]
```

Comment:
The form `let _ = some_must_use();` is preferred as a reader can know where an ignored value is.